corehq/apps/reports/v2/views.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import json
+
+from django.http import HttpResponse
+from django.views.decorators.http import require_POST
+
+from corehq.apps.domain.decorators import login_and_domain_required
+from corehq.apps.reports.v2.reports import get_report
+
+
+@login_and_domain_required
+@require_POST
+def endpoint_data(request, domain, report_slug, endpoint_slug):
+
+    report_config = get_report(request, domain, report_slug)
+    endpoint = report_config.get_data_endpoint(endpoint_slug)
```

Comment:
I'm curious what value we get by exposing the endpoint to the caller. As opposed to just asking the report to provide the data. 

I think the composition pattern with the Data provider is a good one, just wondering whether it can be hidden behind the interface to the report?

Reply:
(same comment applies to the filter endpoint)

Reply:
> just wondering whether it can be hidden behind the interface to the report?

Ideally, the goal is to re-use endpoints in other reports. The endpoint outputs a format that the corresponding JS will used based on the report query. I'm still working this through and at this point I am not ready to get into the nitty gritty just yet as I don't have a fully working example of the idea in my head.